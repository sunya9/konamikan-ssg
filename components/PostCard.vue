<template>
  <article class="card">
    <div class="card-image">
      <nuxt-link :to="$resolvePostUrl(post)">
        <div
          class="image-2by1"
          :class="{
            'has-background-white-bis no-image has-text-grey': !post.feature_image
          }"
        >
          <img
            v-if="post.feature_image"
            class="image-2by1-inner"
            :src="post.feature_image"
            alt=""
          />
        </div>
      </nuxt-link>
    </div>
    <div class="card-content">
      <div class="content">
        <h3 class="title is-5">
          <nuxt-link :to="$resolvePostUrl(post)">
            {{ post.title }}
          </nuxt-link>
        </h3>
        <h4 class="subtitle is-6 has-text-grey">
          <small>{{ $dayjs(post.published_at).format('YYYY/MM/DD') }}</small>
        </h4>
        <p>
          {{ excerpt }}
        </p>
      </div>
    </div>
  </article>
</template>
<script lang="ts">
import Vue, { PropOptions } from 'vue'
import { PostOrPage } from '@tryghost/content-api'

export default Vue.extend({
  props: {
    post: {
      type: Object,
      required: true
    } as PropOptions<PostOrPage>
  },
  computed: {
    excerpt(): string {
      return this.post.custom_excerpt || this.post.excerpt || ''
    }
  }
})
</script>
<style scoped>
.image-2by1 {
  position: relative;
  overflow: hidden;
}

.image-2by1::before {
  content: '';
  display: block;
  padding-top: 50%;
}
.image-2by1-inner {
  position: absolute;
  top: 0;
  right: 0;
  left: 0;
  bottom: 0;
  margin: auto;
  height: auto;
  max-width: 100%;
}
.no-image::after {
  content: 'No image';
  display: inline-block;
  position: absolute;
  top: 0;
  right: 0;
  left: 0;
  bottom: 0;
  margin: auto;
  display: flex;
  justify-content: center;
  align-items: center;
}
</style>
