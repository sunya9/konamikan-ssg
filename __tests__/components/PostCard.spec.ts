import { mount } from '@vue/test-utils'
import { post, postWithFeatureImage } from '@/__tests__/fixtures/postOrPage'
import PostCard from '@/components/PostCard.vue'

describe('PostCard', () => {
  it('normal', () => {
    expect(
      mount(PostCard, {
        propsData: {
          post
        }
      })
    ).toMatchSnapshot()
  })
  it('with feature image', () => {
    expect(
      mount(PostCard, {
        propsData: {
          post: postWithFeatureImage
        }
      })
    ).toMatchSnapshot()
  })
})
