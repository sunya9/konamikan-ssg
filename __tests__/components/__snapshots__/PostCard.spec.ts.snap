// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostCard normal 1`] = `
<article class="card">
  <div class="card-image"><a>
      <div aria-labelledby="__test__title" aria-describedby="__test__description" class="image-2by1 has-background-light no-image has-text-secondary">
        <!---->
      </div>
    </a></div>
  <div class="card-content">
    <div class="content">
      <h3 id="__test__title" class="title is-5"><a>
          title
        </a></h3>
      <h4 class="subtitle is-6 has-text-secondary"><small>
          2020/04/05
        </small></h4>
      <p id="__test__description">
        excerpt
      </p>
    </div>
  </div>
</article>
`;

exports[`PostCard with feature image 1`] = `
<article class="card">
  <div class="card-image"><a>
      <div aria-labelledby="__test__title" aria-describedby="__test__description" class="image-2by1"><img src="feature_image.png" alt="" class="image-2by1-inner"></div>
    </a></div>
  <div class="card-content">
    <div class="content">
      <h3 id="__test__title" class="title is-5"><a>
          title
        </a></h3>
      <h4 class="subtitle is-6 has-text-secondary"><small>
          2020/04/05
        </small></h4>
      <p id="__test__description">
        excerpt
      </p>
    </div>
  </div>
</article>
`;
